utility_analysis_new/combiners.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2022 OpenMined.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""UtilityAnalysisCountCombiner."""
+
+from dataclasses import dataclass
+from typing import List, Sized
+import numpy as np
+
+import pipeline_dp
+from pipeline_dp import dp_computations
+
+
+@dataclass
+class CountUtilityAnalysisMetrics:
+    """ Stores metrics for the count utility analysis.
+
+    Args:
+        count: the number of contributions per a partition.
+        per_partition_contribution_error: the amount of contribution error in a partition.
+        expected_cross_partition_error: the estimated amount of error across partitions.
+        var_cross_partition_error: the variance of the contribution bounding error.
+        std_cb: the standard deviation of the var_cross_partition_error.
+        std_noise: the noise standard deviation for DP count.
+        noise_kind: the type of noise used.
+    """
+    count: int
+    per_partition_contribution_error: int
+    expected_cross_partition_error: float
+    var_cross_partition_error: float
+    std_cb: float
+    std_noise: float
+    noise_kind: pipeline_dp.NoiseKind
+
+
+@dataclass
+class AccumulatorMetrics:
+    count: int
+    per_partition_contribution_error: int
+    expected_cross_partition_error: float
+    var_cross_partition_error: float
+
+
+class UtilityAnalysisCountCombiner(pipeline_dp.Combiner):
+    """ A combiner for utility analysis counts. """
+    AccumulatorType = AccumulatorMetrics
+
+    def __init__(self, params: pipeline_dp.combiners.CombinerParams):
+        self._params = params
+
+    def create_accumulator(self, data: Sized) -> AccumulatorType:
+        """ Create an accumulator for a user and a partition.
+
+        Args:
+            data: contains two columns; 1) a column with the total number of partitions a user contributed to
```

Comment:
Nit: data is tuple, no need to talk about column. something like data is a tuple of ...